CHANGELOG_UNRELEASED.md

```diff
@@ -109,6 +109,31 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
   + we provide an equality of spaces `eqmx_col` between `\mxcol_i V_
     i` and the sum of spaces `\sum_i <<V_ i>>)%MS`.
 
+- new file `mxred.v` which contains the theory of diagonalization. To
+  that effect, we define `conjmx`, `restrictmx`, `similar_to`,
+  `similar`, `similar_in`, `similar_in_to`, `all_similar_to`,
+  `similar_diag`, `diagonalizable_in`, `diagonalizable`,
+  `codiagonalizable_in`, `codiagonalizable`, `similar_trig`,
+  `trigonalizable_in`, `trigonalizable`, `cotrigonalizable_in`, and
+  `cotrigonalizable`; and their theory: `stablemx_comp`,
+  `stablemx_restrict`, `conjmxM`, `conjMmx`, `conjuMmx`, `conjMumx`,
+  `conjuMumx`, `conjmx_scalar`, `conj0mx`, `conjmx0`, `conjumx`,
+  `conj1mx`, `conjVmx`, `conjmxK`, `conjmxVK`, `horner_mx_conj`,
+  `horner_mx_uconj`, `horner_mx_uconjC`, `mxminpoly_conj`,
+  `mxminpoly_uconj`, `sub_kermxpoly_conjmx`, `sub_eigenspace_conjmx`,
+  `eigenpoly_conjmx`, `eigenvalue_conjmx`, `conjmx_eigenvalue`,
+  `similarPp`, `similarW`, `similarP`, `similarRL`, `similarLR`,
+  `similar_mxminpoly`, `similar_diag_row_base`, `similar_diagPp`,
+  `similar_diagP`, `similar_diagPex`, `similar_diagLR`,
+  `similar_diag_mxminpoly`, `similar_diag_sum`, `codiagonalizable1`,
+  `codiagonalizable_on`, `diagonalizable_diag`,
+  `diagonalizable_scalar`, `diagonalizable0`, `diagonalizablePeigen`,
+  `diagonalizableP`, `diagonalizable_conj_diag`, and
+  `codiagonalizableP`.
+
+Remark that even though `trigonalization` is defined, its theory is
```

Comment:
Actually, I do not intend to do any general theory of trigonalization of matrices, however I dealt with trigonalization of complex matrices in [`spectral.v`](https://github.com/math-comp/math-comp/pull/207/files#diff-3921d8b72557faab0a20e006620ea393435554cac831957d8c40bbfa10515519) in the "main PR"... and it would be weird to put the definition there...

Reply:
maybe only add this definition in `mxred`  with the PR that will include `spectral`.